admin-dev/themes/new-theme/js/components/auto-complete-search.js

```diff
@@ -0,0 +1,188 @@
+      display: 'name', // Which field of the object from the list is used for display (can be a string or a callback)
+      value: 'id', // Which field of the object from the list is used for value (can be a string or a callback)
+      limit: 20, // Limit the number of displayed suggestion
+      dataLimit: 0, // How many elements can be selected max
+    };
+    this.dataSetConfig = {...defaultDataSetConfig, ...inputDataSetConfig};
+
+    // Merging object works fine on one level, but on two it erases sub elements even if not present, so
+    // we handle templates separately
+    const defaultTemplates = { // Default rendering functions which can be overridden
+      suggestion(item) {
+        return `<div><img src="${item.image}" style="width:50px" /> ${item.name}</div>`;
+      },
+      pending(query) {
+        return `<div class="px-2">Searching for "${query.query}"</div>`;
+      },
+      notFound(query) {
+        return `<div class="px-2">No results found for "${query.query}"</div>`;
+      },
+      renderSelected: (selectedItem) => `<li>
+            <div>
+              <img src="${selectedItem.image}" style="width:50px" /> ${selectedItem.name}
+            </div>
+          </li>`,
+    };
+    this.dataSetConfig.templates = {...defaultTemplates, ...inputDataSetConfig.templates};
+
+    this.buildTypeahead();
+  }
+
+  /**
+   * Force set values, the selected items are removed Then the component loops through the inputs
+   * and render each object in the list as if it came from the API so the format must match.
+   *
+   * @param values {array}
+   */
+  setValue(values) {
+    this.clearValue();
+    if (!values || values.length <= 0) {
+      return;
+    }
+
+    values.each((value) => {
+      this.appendSelectedItem(value);
+    });
+  }
+
+  /**
+   * Removes selected items.
+   */
+  clearValue() {
+    const formIdItem = $('li', this.$selectionContainer);
+    formIdItem.remove();
+  }
+
+  /**
+   * Build the typeahead component based on provided configuration.
+   */
+  buildTypeahead() {
+    this.$searchInput.typeahead(this.config, this.dataSetConfig)
+      .bind('typeahead:select', (e, selectedItem) => {
+        // When limit is one we cannot select additional elements so we replace them instead
+        // @todo: maybe this behaviour should be defined by an option
+        if (this.dataSetConfig.dataLimit === 1) {
+          this.replaceSelectedItem(selectedItem);
+        } else {
+          this.appendSelectedItem(selectedItem);
+        }
+      }).bind('typeahead:close', (e) => {
+        $(e.target).val('');
+      });
+  }
+
+  /**
+   * When the component is configured to have only one selected element on each selection
+   * the previous selection is removed and then replaced.
+   *
+   * @param selectedItem {Object}
+   * @returns {boolean}
+   */
+  replaceSelectedItem(selectedItem) {
+    this.clearValue();
+    this.addSelectedContentToContainer(selectedItem);
+
+    return true;
```

Comment:
Is it useful to return true?

Reply:
yes because it is used as callback of the `typeahead:select` and much like a click I guess returning `false` prevents some actions after

but you're right to point it out!! because I need to change the initial callback and use `return this. replaceSelectedItem`